src/registry.ts

```diff
@@ -160,31 +160,35 @@ async function processToken(
 ): Promise<string> {
 	const { hostname } = URL.parse(registryBaseUrl);
 	const image = parseImage(imagePath);
-	if (hostname?.endsWith(".docker.io") && !token) {
-		const resp = await dlJson<{ token: string }>(
-			`https://auth.docker.io/token?service=registry.docker.io&scope=repository:${image.path}:pull`,
-			{},
-			allowInsecure,
-		);
-		return `Bearer ${resp.token}`;
-	}
-	if (hostname?.endsWith(".gitlab.com") && token?.startsWith("Basic")) {
-		if (token?.includes(":")) {
-			token = "Basic " + Buffer.from(token?.replace("Basic ", "")).toString("base64");
-		}
-		const resp = await dlJson<{ token: string }>(
-			`https://gitlab.com/jwt/auth?service=container_registry&scope=repository:${image.path}:pull,push`,
-			{ Authorization: token },
-			allowInsecure,
-		);
-		return `Bearer ${resp.token}`;
-	}
+	if (hostname?.endsWith(".docker.io") && !token) return getGitHubToken(image.path, allowInsecure)
```

Comment:
I assume we don't need an await here and two lines down since we already await `processToken` on line 203.

Reply:
This line looks a bit odd. If endsWit docker.io, then get _github_ token?

Reply:
I think it should be getDockerHubToken, not getGitHubToken?

Reply:
My bad! It should of course be **docker**Token.

Reply:
Amended commit to fix the naming error